### PR TITLE
Fix username/password required custom error text for OIE

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
## Description:

Current issue : i18n keys `error.username.required` and `error.password.required` are not applied for v2 (OIE)

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:

Before:

https://user-images.githubusercontent.com/72614880/139138613-53a41950-0a1d-42b3-9778-7e30d76a1635.mov

After:

https://user-images.githubusercontent.com/72614880/139118976-dee70716-56de-414a-82a2-109d5caafdec.mov

### Issue:

- [OKTA-429891](https://oktainc.atlassian.net/browse/OKTA-429891)

### CODE REVIEW

File: src/v2/view-builder/internals/BaseModel.js [hunk 1]
```diff
@@ -25,22 +25,24 @@ const convertUiSchemaFieldToProp = (uiSchemaField) => {
...
-      }
-    };
-  } else if (uiSchemaField.name === 'credentials.passcode' && config.required) {
-    config.required = false;
-    config.validate = function(value) {
-      if (_.isEmpty(value)) {
-        return loc(passwordRequiredi18nKey, 'login');
-      }
-    };
+  if (remediationName === 'identify') {
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/internals/BaseModel.js:28]: base model should not be aware of form names..

**Review by pradeepdewda-okta — CHANGES_REQUESTED** [review]
We should not include form name specific code to BaseModel..

File: test/unit/spec/v2/view-builder/views/IdentifierView_spec.js [hunk 2]
```diff
@@ -201,4 +211,51 @@ describe('v2/view-builder/views/IdentifierView', function() {
...
+        'params':  {
+          'showPasswordToggle': false,
+        },
+        'secret': true,
+        'type': 'password',
+      }]
+    };
+
+    testContext.init(XHRIdentifyWithPassword.remediation.value);
+    testContext.view.form.__save();
```
[the remark this answers is not in the document]
Comment (denysoblohin-okta) [reply test/unit/spec/v2/view-builder/views/IdentifierView_spec.js]: Replaced with `testContext.view.form.isValid()`

File: test/testcafe/spec/Identify_spec.js [hunk 3]
```diff
@@ -150,6 +151,40 @@ test.requestHooks(identifyMockWithUnsupportedResponseError)('should show error i
...
+
+  await identityPage.clickNextButton();
+  await identityPage.waitForErrorBox();
+
+  await t.expect(identityPage.hasIdentifierError()).eql(true);
+  await t.expect(identityPage.hasIdentifierErrorMessage()).eql(true);
+  await t.expect(identityPage.getIdentifierErrorMessage()).eql('Username is required!');
+});
+
+test.requestHooks(identifyRequestLogger, identifyMock)('should be able to submit identifier if error text for password is customized', async t => {
```
[the remark this answers is not in the document]
Comment (denysoblohin-okta) [reply test/testcafe/spec/Identify_spec.js:172]: This test protects from a possible mistake in `validate` function (as in this [snippet](https://github.com/okta/okta-signin-widget/pull/2233#issuecomment-953333352)):
If there is no password value, and it's not needed (no remediation for credentials, only for identifier), validation should pass successfully without adding error to password field.

[the remark this answers is not in the document]
Comment (pradeepdewda-okta) [reply test/testcafe/spec/Identify_spec.js:172]: update the test name to `should not show custom error if password doesn't exist in remediation`

**Review by pradeepdewda-okta — APPROVED** [review]
Looks good now. ready to go. 🚢

**Pull request closed without merging**